rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -425,7 +430,9 @@ def _parse_slot(self, step: Dict[Text, Any]) -> None:
                 for key, value in slot.items():
                     self._add_event(SlotSet.type_name, {key: value})
             elif isinstance(slot, str):
-                self._add_event(SlotSet.type_name, {slot: None})
+                self._add_event(
+                    SlotSet.type_name, {slot: self._slot_default_value(slot)}
```

Comment:
So this is added every time, even if the `slot` has a value?

Reply:
If the slot is referenced by name only, it can't have a value at the same time. If `isinstance(slot, dict)` then we take the given value

Reply:
Gotcha 👍